pkg/minishift/openshift/registry.go

```diff
@@ -25,6 +25,44 @@ import (
 )
 
 func GetDockerRegistryInfo() (string, error) {
+	var registryInfo string
+	var err error
+	if isRegistryRouteEnable() {
+		registryInfo, err = fetchRegistryRoute()
+	} else {
+		registryInfo, err = fetchRegistryService()
+	}
+	return registryInfo, err
+}
+
+func isRegistryRouteEnable() bool {
```

Comment:
This func is imo problematic. There are wrong assumptions about cause and effect. You use _oc_ to see whether there is a route _docker-registry_ and then you basically conclude that this route is created by the add-on and hence later on you append the '443' port. What if the user exposed the route in a different way by manual configuration? 

I think I would feel better if we really check the add-on here, similar to what we do for persistent volumes and anyuid - https://github.com/minishift/minishift/blob/e65bd066c9ee5012abbf0cf162e297967bd45342/pkg/minishift/clusterup/clusterup.go#L211-L211. WDYT?